registry/repository_test.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/sync/errgroup"
+	"oras.land/oras-go/v2/content/memory"
+	"oras.land/oras-go/v2/internal/spec"
+)
+
+// testStorage implements content.ReadOnlyGraphStorage
+type testStorage struct {
+	store *memory.Store
+}
+
+func (s *testStorage) Push(ctx context.Context, expected ocispec.Descriptor, reader io.Reader) error {
+	return s.store.Push(ctx, expected, reader)
+}
+
+func (s *testStorage) Fetch(ctx context.Context, target ocispec.Descriptor) (io.ReadCloser, error) {
+	return s.store.Fetch(ctx, target)
+}
+
+func (s *testStorage) Exists(ctx context.Context, target ocispec.Descriptor) (bool, error) {
+	return s.store.Exists(ctx, target)
+}
+
+func (s *testStorage) Predecessors(ctx context.Context, node ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+	return s.store.Predecessors(ctx, node)
+}
+
+func TestReferrers(t *testing.T) {
```

Comment:
We should also cover test cases where:
1. The input descripor is not a manifest
2. The storage is a `ReferrersLister`
3. The predecessor is of other media type

See https://app.codecov.io/gh/oras-project/oras-go/pull/659/blob/registry/repository.go for the current coverage.

Reply:
Added these three test cases.